feature/1

BahngJH opened the pull request:
Run api 명세 & 작성법 초기 가이드입니다.

BahngJH committed:
서버포트 변경: 8080 > 8081

BahngJH committed:
롬복 의존성 추가

BahngJH committed:
명칭 수정: MemberController > MemberApi

BahngJH committed:
디렉토리 구조 변경
-글로벌(공통)과 도메인으로 나누기 위함

BahngJH committed:
Run api 명세 && 작성법 초기 가이드

BahngJH committed:
Run api 명세 && 작성법 초기 가이드 - 패턴 변경

Soobinnn commented on src/main/java/com/wancs/battle_run/domain/run/api/RunApi.java:
```diff
@@ -0,0 +1,153 @@
+            @ApiResponse(responseCode = "500", description = "INTERNAL SERVER ERROR")
+    })
+    @PostMapping(value = "/save")
+    public ResponseEntity<RecordResponseDto> saveRecord(SaveRecordRequestDto saveRecordRequestDto) throws Exception {
```
@BahngJH  throws Exception을 반복해서 추가하는것보다 RestControllerAdvice 어노테이션을 쓰는건어떨까요

BahngJH replied:
@Soobinnn 수정해서 올렸습니다.

그리고 @RestControllerAdvice 
이 방식이 구글링 했을 때 사용 하는게 여러가지라 베이직하게 폼만 만들어봤는데
큰 틀에서 다른 의견있으면 댓글 달아주세요.
세부적인건 회의를 다시 진행해야될거 같아요.

Soobinnn commented on build.gradle:
```diff
@@ -17,6 +17,7 @@ repositories {
 dependencies {
 	implementation 'org.springframework.boot:spring-boot-starter-web'
 	implementation 'org.springdoc:springdoc-openapi-ui:1.6.9'
+	implementation 'org.projectlombok:lombok'
```
lombok implementation으로 해도 빌드실행할때 에러안나나요?

Soobinnn replied:
compileOnly 'org.projectlombok:lombok'
	annotationProcessor 'org.projectlombok:lombok'
	
	선언해줘야할거같은데용

BahngJH replied:
@Soobinnn 구글링해보니 다른 의존성과는 다르게 그냥 주입만해서는 안되고
어노테이션프로세싱이 필요하다고 해서 수정했습니다.

Soobinnn commented on src/main/java/com/wancs/battle_run/domain/run/api/RunApi.java:
```diff
@@ -0,0 +1,153 @@
+    })
+    @PostMapping(value = "/customSave")
+    public ResponseEntity<RecordResponseDto> customSaveRecord(SaveRecordRequestDto saveRecordRequestDto) throws Exception {
+        HttpHeaders headers = new HttpHeaders();
```
header도 항상 호출하는게 아니라 뭔가 ResponseEnttiy 공통 구현체로 처리해주고싶은데 이건 이슈 만들어서 따로 고민해서 처리해보죵

Soobinnn replied:
해당건은 https://github.com/wa-ncs/battle-run-api/issues/6 에서 처리하는걸로

BahngJH replied:
@Soobinnn Response format 재정의 했는데 확인 부탁드립니다.

Soobinnn commented:
위 코멘트에서 남긴것 처럼 response format은
![image](https://user-images.githubusercontent.com/40817586/186942874-eeefee78-4907-431f-a315-834d54e9c694.png)
이것도 https://github.com/wa-ncs/battle-run-api/issues/6 여기서 처리하는걸로

Soobinnn commented on src/main/java/com/wancs/battle_run/domain/run/dto/RecordResponseDto.java:
```diff
@@ -0,0 +1,24 @@
+    private int calorie;
+    private String imageUrl;
+
+    @Builder
```
흠,, Builder패턴이 사실상 AllArgsConstructor를 사용하는건데,, 아무리 생각해도 DTO에 빌더패턴을 사용안하는게 좋을거같아요 정적팩터리 메서드를 사용해서 .from으로 생성하는게 젤 좋을 것같습니당. 자세한건 제가 예제로 만들게요

BahngJH replied:
@Soobinnn  샘플 받으면 수정 예정

BahngJH committed:
롬복 의존성 수정

BahngJH committed:
@RestControllerAdvice 적용

BahngJH committed:
Response format 재정의

Soobinnn commented on src/main/java/com/wancs/battle_run/domain/run/api/RunApi.java:
```diff
@@ -0,0 +1,144 @@
+            @ApiResponse(responseCode = "500", description = "INTERNAL SERVER ERROR")
+    })
+    @PostMapping(value = "/customSave")
+    public ResponseEntity<CommonResponseDto<RecordResponseDto>> customSaveRecord(SaveRecordRequestDto saveRecordRequestDto) {
```
수기 운동기록 저장하고 기록저장 API가 나눠지는이유가있나요? 동일한 RequestDTO같은데욤

Soobinnn committed:
1. API 명세 url restful 하도록 정리
2. runAPI -> record API 로 명칭 변경 ( url /api/records로 변경)
3. dto request, response 디렉터리(패키지) 분리
4. commonResponseDto -> responseDTO로 명칭변경, global로 위치이동
5. apiprefix 선언

Soobinnn commented:
before
![image](https://user-images.githubusercontent.com/40817586/187040892-d64009c9-edf8-43ff-b28f-9eee68e14972.png)

after
![image](https://user-images.githubusercontent.com/40817586/187041793-ce99c842-c206-46ad-93a3-6e06d5caa725.png)

Soobinnn committed:
url 오기 수정

Soobinnn committed:
1..메소드명 변경,
2. records 수정 API path param 추가
3. 운동기록 조회 API 추가

Soobinnn committed:
전체 운동 기록 조회 -> 누적 운동 기록 조회 로  설명 변경

Soobinnn committed:
/api url 수정 , description 수정

Soobinnn commented:
논의한 대로 api url, 명세 수정 후 머지합니다
